lib/labels_web/controllers/page_controller.ex

```diff
@@ -7,4 +7,59 @@ defmodule LabelsWeb.PageController do
   def index(conn, _params) do
     render(conn, "index.html")
   end
+
+  def sync(conn, %{"sync_labels" => form}) do
+    token = conn.assigns.github_token
+    source_owner = form["source_owner"]
+    source_repo = form["source_repo"]
+    target_owner = form["target_owner"]
+    target_repo = form["target_repo"]
+
+    with {:ok, source_labels, target_labels} <-
+           get_labels(token, source_owner, source_repo, target_owner, target_repo),
+         :ok <-
+           create_update_labels(token, target_owner, target_repo, source_labels, target_labels) do
+      conn
+      |> put_flash(:info, "Labels synced!")
+      |> redirect(to: Routes.page_path(conn, :index))
+    else
+      {:error, :not_found, repo} ->
+        conn
+        |> put_flash(:error, "#{repo} repository not found")
+        |> redirect(to: Routes.page_path(conn, :index))
+    end
+  end
+
+  # get source and target labels
+  defp get_labels(token, source_owner, source_repo, target_owner, target_repo) do
+    with {{:ok, source_labels}, :source} <-
+           {github_api().get_labels(token, source_owner, source_repo), :source},
+         {{:ok, target_labels}, :target} <-
+           {github_api().get_labels(token, target_owner, target_repo), :target} do
+      {:ok, source_labels, target_labels}
+    else
+      {{:error, :not_found}, repo} -> {:error, :not_found, repo}
+    end
+  end
+
+  # Send request to update or create labels
+  defp create_update_labels(token, owner, repo, source_labels, target_labels) do
+    target_label_names = Enum.map(target_labels, & &1["name"])
+
+    source_labels
+    |> Enum.map(fn label ->
+      if Enum.member?(target_label_names, label["name"]) do
```

Comment:
If the `source` label name already exists on the `target` repository we send an `update` API request to Github otherwise we send a `create` request